recommendation/owin.json

```diff
@@ -0,0 +1,478 @@
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "Add a comment to explain how to use FileServer and to ensure the usage of FileProvider attribute\r\n",
+          "Actions": [
+            {
+              "Name": "AddComment",
+              "Type": "Method",
+              "Value": "Please replace the FileSystem property inside of the new FileServerOptions with FileProvider instead, if FileSystem is not initialized please add FileProvider.",
+              "Description": "Add a comment on how to setup your MVC controller in startup class.",
+              "ActionValidation": {
+                "Contains": "FileProvider",
+                "NotContains": ""
+              }
+            }
+          ]
+        }
+      ]
+    },
+    {
+      "Type": "Method",
+      "Name": "UseDirectoryBrowser",
+      "Value": "Owin.IAppBuilder.UseDirectoryBrowser(Microsoft.Owin.StaticFiles.DirectoryBrowserOptions)",
+      "KeyType": "Name",
+      "ContainingType": "DirectoryBrowserExtensions",
+      "RecommendedActions": [
+        {
+          "Source": "Amazon",
+          "Preferred": "Yes",
+          "TargetFrameworks": [
+            {
+              "Name": "netcoreapp3.1",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            },
+            {
+              "Name": "net5.0",
+              "TargetCPU": [ "x86", "x64", "ARM32", "ARM64" ]
+            }
+          ],
+          "Description": "Replace UseErrorPage with UseErrorPage\r\n",
+          "Actions": [
+            {
+              "Name": "AddComment",
+              "Type": "Method",
+              "Value": "Please replace the FileSystem property inside of the new DirectoryBrowsingOptions with FileProvider instead, if FileSystem is not initialized please add FileProvider.",
+              "Description": "Add a comment on how to setup your MVC controller in startup class.",
+              "ActionValidation": {
+                "Contains": "FileProvider",
+                "NotContains": ""
+              }
+            },
+            {
+              "Name": "AddDirective",
+              "Type": "Using",
+              "Value": "Microsoft.Extensions.DependencyInjection",
+              "Description": "Add Microsoft.AspNetCore.Builder namespace",
+              "ActionValidation": {
+                "Contains": "usingMicrosoft.Extensions.DependencyInjection",
+                "NotContains": ""
+              }
+            },
+            {
+              "Name": "AddComment",
+              "Type": "Method",
+              "Value": "Please add a new ConfigureServices method: public void ConfigureServices(IServiceCollection services) { services.AddDirectoryBrowser(); }",
```

Comment:
handling via custom rules ? (check the feasibility)

Reply:
Yup I made notes of all these gaps and will create tickets for all of them when the Owin rules are in. Ultimately we want all comments changed into actual rules instead.